### ISSUE TITLE
Incorrect unreferenced_argument errors

### ISSUE THREAD

**Issue by rysenko** [joyent/javascriptlint#14]
_Code example:_

```
var loadData = function (event, context) {
    if (context && context.hasActionChanged()) {
        return;
    }};
}
```

_Actual result:_ jsLint gives erroneous warning for the first argument of the function, if it's not used and  it's not last. See below:
{code}/details.js(127): warning: argument declared but never referenced: event  (unreferenced_argument){code}

_Expected result:_ jsLint should give a warning only for the arguments that are not used and are located after the last argument, which is used.
